src/db_adapters/accounts.rs

```diff
@@ -0,0 +1,150 @@
+use diesel::r2d2::{ConnectionManager, Pool};
+use diesel::{ExpressionMethods, PgConnection};
+use futures::join;
+use tokio_diesel::AsyncRunQueryDsl;
+use tracing::error;
+
+use near_indexer::near_primitives;
+
+use crate::models;
+use crate::schema;
+
+/// Saves new Accounts to database or deletes the ones should be deleted
+pub(crate) async fn handle_accounts(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    outcomes: &near_indexer::ExecutionOutcomesWithReceipts,
+) {
+    let successful_outcomes: Vec<near_indexer::IndexerExecutionOutcomeWithReceipt> = outcomes
+        .values()
+        .filter(|outcome_with_receipt| {
+            match outcome_with_receipt.execution_outcome.outcome.status {
+                near_primitives::views::ExecutionStatusView::SuccessValue(_)
+                | near_primitives::views::ExecutionStatusView::SuccessReceiptId(_) => true,
+                _ => false,
+            }
+        })
+        .cloned()
+        .collect();
+
+    let store_accounts_future = store_accounts(&pool, &successful_outcomes);
+    let remove_accounts_future = remove_accounts(&pool, &successful_outcomes);
+
+    join!(store_accounts_future, remove_accounts_future);
+}
+
+async fn store_accounts(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    outcomes: &[near_indexer::IndexerExecutionOutcomeWithReceipt],
+) {
+    let accounts_to_create: Vec<models::accounts::Account> = outcomes
+        .iter()
+        .filter_map(|outcome_with_receipt| {
+            if let Some(receipt) = &outcome_with_receipt.receipt {
+                match &receipt.receipt {
+                    near_primitives::views::ReceiptEnumView::Action { actions, .. } => {
+                        let accounts: Vec<models::accounts::Account> = actions
+                            .iter()
+                            .filter_map(|action| {
+                                if let near_primitives::views::ActionView::CreateAccount = action {
+                                    Some(models::accounts::Account::new(
+                                        receipt.receiver_id.to_string(),
+                                        &receipt.receipt_id,
+                                    ))
+                                } else {
+                                    None
+                                }
+                            })
+                            .collect();
+                        Some(accounts)
+                    }
+                    _ => None,
+                }
+            } else {
+                None
+            }
+        })
+        .flatten()
+        .collect();
+
+    loop {
+        match diesel::insert_into(schema::accounts::table)
+            .values(accounts_to_create.clone())
+            .on_conflict_do_nothing()
```

Comment:
If there is an existing account, we should update its `created_by_receipt_id`. (e.g. in case CreateAccount -> DeleteAccount -> CreateAccount we should arrive with Account holding the last receipt_id in `created_by_receipt_id` and `deleted_by_receipt_id` should be NULL, I suppose)